client/client.go

```diff
@@ -920,6 +928,13 @@ type MetricsTimer = metrics.Timer
 // MetricsNopHandler is a noop handler that does nothing with the metrics.
 var MetricsNopHandler = metrics.NopHandler
 
+// NewUpdateWorkflowOperation returns an UpdateWorkflowOperation to perform Update-with-Start,
+// or an error in case the update operation is invalid. After executing Client.ExecuteWorkflow with the
+// UpdateWorkflowOperation in the start options, the update result can be obtained.
+func NewUpdateWorkflowOperation(options UpdateWorkflowOptions) (*UpdateWorkflowOperation, error) {
```

Comment:
Want to close on this concern. This is a generically named thing and I think it'd be more reasonable to be named similar to the field/type how/where it is used. This name makes one think this is the common update workflow operation function.

Reply:
If we did do multiop in the future would we reuse `UpdateWorkflowOperation`? I don't see any reason why we wouldn't so I have no issue with the current name.

Reply:
I don't think we should assume that would be the case, but even if it was, this is too generically named even for that IMO.